src/services/platform.ts

```diff
@@ -1,20 +1,25 @@
 import { utils } from 'ethers';
 
-import { IPlatform } from '../models/Models';
+import { AssetType, IPlatform } from '../models/Models';
 import Platform from '../models/PlatformSchema';
 import { createLogger } from '../logger';
 
 const logger = createLogger('platformService');
 
-async function matchTxToPlatform(tx: string): Promise<IPlatform> {
+async function matchTxToPlatform(tx: string, assetType?: AssetType): Promise<IPlatform> {
   try {
     const parsed = utils.parseTransaction(tx);
     const to = parsed.to.toLowerCase();
     const data = parsed.data.toLowerCase();
     const platforms = await Platform.find();
-
     for (const platform of platforms) {
-      for (const contract of platform.whitelist) {
+      const whitelistAddresses = platform['whitelist'][assetType][parsed.chainId];
```

Comment:
This will not properly handle the case when there's no db entry for a given combination of `assetType` and `chainId`. The simplest thing you can do is `platform['whitelist'][assetType]?.[parsed.chainId] || []`
That way this will never error out